strawberry/codegen/query_codegen.py

```diff
@@ -487,6 +490,28 @@ def _convert_operation(
         variables, variables_type = self._convert_variable_definitions(
             operation_definition.variable_definitions, operation_name=operation_name
         )
+        if variables_type:
+            # There are multiple nodes in the `operation_definition.variable_definitions`
+            # list and collectively they are used to make the `variables_type`, so the
+            # closest node to the variable definition is the full operation definition node.
+            if variables_type.graphql_node is not None:
+                raise ValueError(
+                    "Internal codegen error.  graphql_node should NOT be set."
+                )
+            variables_type.graphql_node = operation_definition
+
+            # There isn't actually a class in the strawberry schema that represents the
+            # variables in a query (it's _usually_ a wrapper-class around some
+            # `@strawberry.input` types that are referenced in the resolver function's
+            # arguments, however, the query variable could also be an input to some
+            # directive).  To this end, we currently just use the containing operation type
+            # (e.g. the ``Query`` or ``Mutation`` class and then we'll let consumers sort
+            # out how they want to use this metadata).
+            if variables_type.graphql_type is not None:
+                raise ValueError(
```

Comment:
Yes, this is defensive to make sure we didn't mess something up.

These are the `{Type}Variables` objects that I mentioned in the other comments.  Basically, at the client level, it's just a dataclass that describes the inputs to a strawberry resolver (rather than being an actual _type_ on the strawberry/python side).

In that case, I made the decision to just set the graphql node to the `Operation` node and I set the graphql type to the query/mutation type.